sentry_sdk/profiler.py

```diff
@@ -447,13 +489,20 @@ def _sample_stack(*args, **kwargs):
             This should be called at a regular interval to collect samples.
             """
 
-            self.write(
-                nanosecond_time(),
-                [
-                    (str(tid), extract_stack(frame))
-                    for tid, frame in sys._current_frames().items()
-                ],
-            )
+            now = nanosecond_time()
+            raw_sample = {
+                tid: extract_stack(frame, cwd, last_sample[0].get(tid))
+                for tid, frame in sys._current_frames().items()
+            }
+
+            last_sample[0] = raw_sample
+
+            sample = [
+                (str(tid), (stack_id, stack))
```

Comment:
Moving this `str` conversion here was actually a minor optimization. The same thread id gets reused across multiple profiles if they cover the same sample. So if we moved it, we'd be doing this conversion once per profile instead of once in total.